yarn-project/prover-node/src/prover-node.ts

```diff
@@ -65,29 +72,35 @@ export class ProverNode {
       this.l2BlockSource.getProvenBlockNumber(),
     ]);
 
-    if (latestProvenBlockNumber >= latestBlockNumber) {
-      this.log.debug(`No new blocks to prove`, { latestBlockNumber, latestProvenBlockNumber });
+    // Consider both the latest block we are proving and the last block proven on the chain
+    const latestBlockBeingProven = this.latestBlockWeAreProving ?? 0;
+    const latestProven = Math.max(latestBlockBeingProven, latestProvenBlockNumber);
+    if (latestProven >= latestBlockNumber) {
+      this.log.debug(`No new blocks to prove`, { latestBlockNumber, latestProvenBlockNumber, latestBlockBeingProven });
       return;
     }
 
-    const fromBlock = latestProvenBlockNumber + 1;
+    const fromBlock = latestProven + 1;
     const toBlock = fromBlock; // We only prove one block at a time for now
-    await this.prove(fromBlock, toBlock);
+
+    await this.startProof(fromBlock, toBlock);
+    this.latestBlockWeAreProving = toBlock;
   }
 
   /**
    * Creates a proof for a block range. Returns once the proof has been submitted to L1.
    */
-  public prove(fromBlock: number, toBlock: number) {
-    return this.createProvingJob().run(fromBlock, toBlock);
+  public async prove(fromBlock: number, toBlock: number) {
+    const job = await this.createProvingJob(fromBlock);
+    return job.run(fromBlock, toBlock);
   }
 
   /**
    * Starts a proving process and returns immediately.
    */
-  public startProof(fromBlock: number, toBlock: number) {
-    void this.createProvingJob().run(fromBlock, toBlock);
-    return Promise.resolve();
+  public async startProof(fromBlock: number, toBlock: number) {
+    const job = await this.createProvingJob(fromBlock);
+    void job.run(fromBlock, toBlock);
```

Comment:
IIUC this is only public for testing reasons? Otherwise an instance that proves automatically could start an auto proof at the same time since this doesn't block `work`

Reply:
The idea is exposing both prove and startProof on the public API, one of them blocking until proof is submitted and the other not. To be honest I'm not sure which one we'll end up using for prover-sequencer coordination, but I wanted to keep both for the time being.